src/explore-education-statistics-frontend/src/middleware/pages/redirectPages.ts

```diff
@@ -19,66 +19,68 @@ const redirectPaths = {
   publications: '/find-statistics',
 };
 
-export default async function redirectPages(request: NextRequest) {
-  const { nextUrl } = request;
-  const decodedPathname = decodeURIComponent(request.nextUrl.pathname);
+export default function redirectPages(middleware: NextMiddleware) {
+  return async (request: NextRequest, event: NextFetchEvent) => {
```

Comment:
Think we could make this a bit flatter and more Express-like by not making this a curried function. 

We could change the middleware function signature to:

```ts
export default async function redirectPages(
  request: NextRequest,
  event: NextFetchEvent,
  middleware: NextMiddleware,
) {
  // Rest of code...
}
```

As we don't need to return a function anymore, we avoid the extra indentation and don't need to change the entire function.

We just need to make some tweaks to `chain.ts`:

```ts
import { NextMiddlewareResult } from 'next/dist/server/web/types';
import {
  NextFetchEvent,
  NextMiddleware,
  NextRequest,
  NextResponse,
} from 'next/server';

export type ChainedMiddleware = (
  request: NextRequest,
  event: NextFetchEvent,
  next: NextMiddleware,
) => NextMiddlewareResult | Promise<NextMiddlewareResult>;

export default function chain(
  middlewares: ChainedMiddleware[],
): NextMiddleware {
  const [current, ...remaining] = middlewares;

  if (current) {
    const next = chain(remaining);
    return (request, event) => current(request, event, next);
  }

  return () => NextResponse.next();
}
```

Note we don't need to expose an `index` parameter as we can just take the first element from the array.

This also needs a slight tweak to the middleware test function:

```ts
export default function middlewareTestRun(
  middleware: ChainedMiddleware,
  page: string,
) {
  const request = new NextRequest(new Request(page));

  const fetchEvent = new NextFetchEvent({
    page,
    request,
  });

  return middleware(request, fetchEvent, () => NextResponse.next());
}
```